pkg/comp-functions/functions/vshnnextcloud/deploy.go

```diff
@@ -0,0 +1,395 @@
+	if err != nil {
+		return err
+	}
+
+	return svc.SetDesiredComposedResource(pg)
+}
+
+func addRelease(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNNextcloud, adminSecret string) error {
+	release, err := newRelease(ctx, svc, comp, adminSecret)
+	if err != nil {
+		return err
+	}
+
+	return svc.SetDesiredComposedResourceWithName(release, comp.GetName()+"-release")
+}
+
+func getResources(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNNextcloud) (common.Resources, error) {
+	l := svc.Log
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+
+	resources, err := utils.FetchPlansFromConfig(ctx, svc, plan)
+	if err != nil {
+		err = fmt.Errorf("cannot fetch plans from the composition config, maybe they are not set: %w", err)
+		return common.Resources{}, err
+	}
+
+	res, errs := common.GetResources(&comp.Spec.Parameters.Size, resources)
+	if len(errs) != 0 {
+		l.Error(errors.Join(errs...), "Cannot get Resources from plan and claim")
+	}
+	return res, nil
+}
+
+func newValues(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNNextcloud, adminSecret string) (map[string]any, error) {
+	values := map[string]any{}
+
+	res, err := getResources(ctx, svc, comp)
+	if err != nil {
+		return nil, err
+	}
+
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+	nodeSelector, err := utils.FetchNodeSelectorFromConfig(ctx, svc, plan, comp.Spec.Parameters.Scheduling.NodeSelector)
+	if err != nil {
+		return values, fmt.Errorf("cannot fetch nodeSelector from the composition config: %w", err)
+	}
+
+	externalDb := map[string]any{}
+
+	if !comp.Spec.Parameters.Service.DefaultInternalDB {
+		cd, err := svc.GetObservedComposedResourceConnectionDetails(comp.GetName() + pgInstanceNameSuffix)
+		if err != nil {
+			return nil, err
+		}
+		externalDb = map[string]any{
+			"enabled": true,
+			"type":    "postgresql",
+			"existingSecret": map[string]any{
+				"enabled":     true,
+				"secretName":  pgSecretName,
+				"usernameKey": "POSTGRESQL_USER",
+				"passwordKey": "POSTGRESQL_PASSWORD",
+				"hostKey":     "POSTGRESQL_HOST",
+				"databaseKey": "POSTGRESQL_DB",
+			},
+			"host":     string(cd[vshnpostgres.PostgresqlHost]) + ":" + string(cd[vshnpostgres.PostgresqlPort]),
+			"database": string(cd[vshnpostgres.PostgresqlDb]),
+			"user":     string(cd[vshnpostgres.PostgresqlUser]),
+			"password": string(cd[vshnpostgres.PostgresqlPassword]),
+		}
+	}
+
+	values = map[string]any{
+		"nextcloud": map[string]any{
+			"host": comp.Spec.Parameters.Service.FQDN,
+			"existingSecret": map[string]any{
+				"enabled":     true,
+				"secretName":  adminSecret,
+				"usernameKey": adminUserSecretField,
+				"passwordKey": adminPWSecretField,
+			},
+			"configs": map[string]string{
+				"vshn-nextcloud.config.php": nextcloudConfig,
+			},
+			"containerPort":   8080,
+			"securityContext": nil,
+			"podSecurityContext": map[string]any{
+				"runAsUser":                nil,
```

Comment:
Will this work on vanilla k8s? Also it seems we are decreasing the security for k8s.

Reply:
I agree. In `svc.Config` you can access the information if it's OpenShift or not.

Reply:
Refactored the code, to take this into consideration